contracts/most/UtBase.sol

```diff
@@ -0,0 +1,228 @@
+            "Setup is already done."
+        );
+
+        tokenSymbol = TOKEN_SYMBOL;
+        tokenName = TOKEN_NAME;
+        tokenDecimals = TOKEN_DECIMALS;
+        totalTokenSupply = genesisTotalSupply;
+
+        /*
+         * In the genesis block, validators gets base token equivalent to the
+         * amount they deposited before joining the core. Equivalent
+         * amount of ERC20 token should be minted and UtBase contract should
+         * hold that balance. Similar flow will happen in normal mint process
+         * where UtBase contract ERC20 balance will be incremented. So in the
+         * setup, UtBase contract holds the ERC20 token balance equal to the
+         * genesisTotalSupply.
+         */
+        balances[address(this)] = genesisTotalSupply;
+    }
+
+    /**
+     * @notice Mints the base coin equivalent amount to beneficiary address.
+     *
+     * @dev Function requires :
+     *          - Caller must be Consensus cogateway contract.
+     *
+     * @param _beneficiary Beneficairy address which will receive base coin.
+     * @param _amount Amount of UtBase tokens to be minted.
+     *
+     */
+    function mint(
+        address payable _beneficiary,
+        uint256 _amount
+    )
+        external
+        onlyConsensusCogateway
+    {
+        _mint(_beneficiary, _amount);
+        unwrapInternal(_beneficiary, _amount);
+    }
+
+    /**
+     * @notice Unwrap converts ERC20 UtBase to base coin.
+     *
+     * @dev  This contract's base coin balance must always be greater than
+     *       unwrap amount.
+     *
+     * @param _amount Amount of ERC20 UtBase token to convert to base coin.
+     */
+    function unwrap(
+        uint256 _amount
+    )
+        external
+    {
+        unwrapInternal(msg.sender, _amount);
+    }
+
+    /**
+     * @notice Wrap converts base coin to ERC20 UtBase.
+     *
+     * @dev The ERC20 OST balance of contract should always be greater than the
+     *      received payable amount.
+     *
+     *      Function requires:
+     *          - msg.value should be non-zero
+     */
+    function wrap()
+        external
+        payable
+    {
+        uint256 amount = msg.value;
+        address account = msg.sender;
+
+        require(
+            amount > 0,
+            "Payable amount should not be zero."
+        );
+
+        assert(balances[address(this)] >= amount);
+
+        transferBalance(address(this), account, amount);
+    }
+
+    /**
+     * @notice It allows to burn UtBase tokens for an account. It calls
+     *         ERC20Token::_burn method.
+     *
+     * @param _account The account whose tokens will be burnt.
+     * @param _value The amount that will be burnt.
+     */
+    function burn(address _account, uint256 _value)
+        external
+    {
+        _burn(_account, _value);
+    }
+
+    /**
+     * @notice It allows to burn tokens of the spender. It calls
+     *         ERC20Token::_burnFrom method.
```

Comment:
Similarly, the internal function is not needed in the documentations.

Reply:
You are correct. Internal details is not required here.
So removed it.